### ISSUE TITLE
Allow tests to be marked as requiring a system/platform/etc pattern file suffix

### ISSUE THREAD

**Issue by zaufi** [zaufi/pytest-matcher#22]
I.e., in the situation when some (not all!) pattern files have some system-specific content (e.g., different EOL styles).

Smth like this:
```python
def generic_test(capfd, expected_out):
    ...
    stdout, _ = capfd.readouterr()
    # Get content from `<base-dir>/.../generic_test.out`
    assert expected_out == stdout

@pytest.mark.expect_system_suffix
def system_specific_test(capfd, expected_out):
    ...
    stdout, _ = capfd.readouterr()
    # Get content from `<base-dir>/.../system_specific_test-Linux.out`
    assert expected_out == stdout
```

**Comment by zaufi** [comment]
Actually, it could be almost anything:
```python
@pytest.mark.expect_system_suffix(
    suffix=platform.system()
  )
def system_specific_test(capfd, expected_out):
    ...
    stdout, _ = capfd.readouterr()
    # Get content from `<base-dir>/.../system_specific-Linux.out`
    assert expected_out == stdout

@pytest.mark.expect_system_suffix(
    suffix=f'py{sys.version_info.major}.{sys.version_info.minor}'
  )
def python_specific_test(capfd, expected_out):
    ...
    stdout, _ = capfd.readouterr()
    # Get content from `<base-dir>/.../python_specific_test-py3.12.out`
    assert expected_out == stdout
```

**Comment by zaufi** [comment]
@xymaxim Ping ;-)

**Comment by xymaxim** [comment]
I like how it's flexible!

**Comment by zaufi** [comment]
Implemented in PR #23
